go.mod

```diff
@@ -1,5 +1,11 @@
 module github.com/honeycombio/dynsampler-go
 
-go 1.12
+go 1.17
```

Comment:
I think it would be best to bump the go version in its own `maint:` PR

Reply:
Fair; I'll want to merge that before this one, then, and rebase this.

Reply:
Agreed